templates/admin/src/app.raw.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from "react";
+import { Theme, withStyles } from "@material-ui/core/styles";
+
+import GenericApp from "@iobroker/adapter-react/GenericApp";
+import Settings from "./components/settings";
+import { GenericAppProps, GenericAppSettings } from "@iobroker/adapter-react/types";
+import { StyleRules } from "@material-ui/styles";
+
+const styles = (_theme: Theme): StyleRules => ({
+	root: {},
+});
+
+class App extends GenericApp {
+	constructor(props: GenericAppProps) {
+		const extendedProps: GenericAppSettings = { ...props };
+		extendedProps.encryptedFields = [];
+		extendedProps.translations = {
+			"en": require("./i18n/en"),
+			"de": require("./i18n/de"),
+			"ru": require("./i18n/ru"),
+			"pt": require("./i18n/pt"),
+			"nl": require("./i18n/nl"),
+			"fr": require("./i18n/fr"),
+			"it": require("./i18n/it"),
+			"es": require("./i18n/es"),
+			"pl": require("./i18n/pl"),
+			"zh-cn": require("./i18n/zh-cn"),
+		};
+
+		super(props, extendedProps);
+	}
+
+	onConnectionReady(): void {
+		// executed when connection is ready
+	}
+
+	render(): React.ReactNode {
```

Comment:
Does ESLint require this type annotation? I like it when TypeScript infers the return type in React.

Reply:
Currently ESLint requires return types to be defined. We could of course change this, but I like it as a way to cross-check that you're returning what you wanted. I agree, for React it would be ok to leave it out, but for all other TS code I wouldn't do that.

Reply:
Agreed. The React types are too confusing IMO to force them.

Reply:
So, what's the conclusion?

Reply:
Keep annotations for plain TS, remove them for React

Reply:
Done.